### PR TITLE
Add setting to only show events starting in x minutes.

### PR DESCRIPTION
**Pull request by KaiOelfke**
### Status
**READY**

### Description
This PR adds a setting to only show events in x minutes. It implements the feature request #176. It's only filtering out future events that start later then the configured threshold. Current or past events are displayed as configured by other settings.

Bildschirmfoto 2021-02-12 um 19 08 41

### Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

1. Have the next event in e.g. 60 minutes.
2. Open Settings and configure it to show only events starting in 30 minutes.

The event should not be displayed in the title.

3. Move event to in 25 minutes.

The event should be now displayed in the title.

### CODE REVIEW

File: MeetingBar/Views/Preferences/AppearanceTab.swift [hunk 1]
```diff
@@ -75,9 +75,7 @@ struct StatusBarSection: View {
             HStack {
                 Picker("Time", selection: $eventTimeFormat) {
                     Text("show").tag(EventTimeFormat.show)
-                    if #available(OSX 11.0, *) {} else {
```
Comment (jgoldhammer) [inline MeetingBar/Views/Preferences/AppearanceTab.swift:78]: issue : Please reintegrate the change from upstream. We want to hide this option in Big Sur...

File: MeetingBar/Views/Preferences/AppearanceTab.swift [hunk 2]
```diff
@@ -140,14 +140,18 @@ struct EventsSection: View {
...
-
-
                 Picker("All day events:", selection: $allDayEvents) {
                     Text("show").tag(AlldayEventsAppereance.show)
                     Text("show only with meeting link").tag(AlldayEventsAppereance.show_with_meeting_link_only)
                     Text("hide").tag(AlldayEventsAppereance.hide)
                 }
             }
+            HStack {
+                Toggle("Show only events starting in", isOn: $showEventMaxTimeUntilEventEnabled)
```
Comment (jgoldhammer) [inline MeetingBar/Views/Preferences/AppearanceTab.swift:150]: **suggestion**: Can we change this to "Show only next event starting within next"? Or what do you think @leits ?

Comment (jgoldhammer) [reply MeetingBar/Views/Preferences/AppearanceTab.swift:150]: @leits question: Currently this option is in the events section. Should we move it to statusbar section to make clear that this only affects the statusbar and not the menu?

**Review by jgoldhammer — CHANGES_REQUESTED** [review]
Overall looks good to me. Please lets discuss my questions .

File: MeetingBar/Views/Preferences/AppearanceTab.swift (hunk 2 above)
Comment (KaiOelfke) [reply MeetingBar/Views/Preferences/AppearanceTab.swift:150]: I agree, it makes more sense in the status bar section and "Show only next event starting within" is clearer copy.

File: MeetingBar/Views/Preferences/AppearanceTab.swift (hunk 1 above)
Comment (KaiOelfke) [reply MeetingBar/Views/Preferences/AppearanceTab.swift:78]: That change wasn't intended. Thanks for the heads up.

**Comment by leits** [comment]
What worries me is that if there are no meetings at the next chosen time, the icon will be the same as when all the meetings are over. What do you think about it, @KaiOelfke, @jgoldhammer ?

**Comment by jgoldhammer** [comment]
> What worries me is that if there are no meetings at the next chosen time, the icon will be the same as when all the meetings are over. What do you think about it, @KaiOelfke, @jgoldhammer ?

That would be indeed a problem for the user. It would mean to read all possible next event candidates from the apple calendar and show a different icon if there is another next event for today or today+tomorrow. Does this makes sense?

**Comment by leits** [comment]
Yes, I think so.
Maybe the app icon?

**Comment by KaiOelfke** [comment]
Maybe, I'll have time this weekend to look at this. There's also the SF Symbol: calendar.badge.clock

**Comment by KaiOelfke** [comment]
I moved the logic to updateTitle and added an enum to handle the different cases. I'm not sure about the title logic. But the icon for the threshold case should be the app icon or `iconCalendar` and not the checkmark one.

File: MeetingBar/AppDelegate.swift [hunk 3]
```diff
@@ -413,7 +427,7 @@ class AppDelegate: NSObject, NSApplicationDelegate, UNUserNotificationCenterDele
             preferencesWindow.close()
         }
         preferencesWindow = NSWindow(
-            contentRect: NSRect(x: 0, y: 0, width: 700, height: 550),
+            contentRect: NSRect(x: 0, y: 0, width: 700, height: 600),
```
Comment (leits) [inline MeetingBar/AppDelegate.swift:430]:
```suggestion
            contentRect: NSRect(x: 0, y: 0, width: 700, height: 550),
```

**Review by leits — APPROVED** [review]

**Pull request merged**